### DISCUSSION TITLE
envchain vs Skate

### DISCUSSION THREAD

**Discussion by ProfessorManhattan** [charmbracelet/skate#69] [Q&A] [answered]
Hey, I'm currently implementing a system for handling secret variables. I'm split between `envchain` and `skate` right now. Does anyone have any opinion about which one is better and pros /cons of each tool?

**Comment by bashbunni** [comment, marked as answer]
It's not clear to me from the `envchain` docs what the encryption is like nor if you can share it across machines. I'd say those are a couple of big pros for `skate` is that everything is end-to-end encrypted without any extra steps from the user. It's also easy to link multiple machines with `charm link` so you can access your skate database from multiple locations. 

One outstanding issue with `skate` right now that I think is important is that there's no support **yet** for encrypted (password protected) SSH keys. I'd love to use a password protected key or even an `ed25519-sk` key where I can use my hardware security key for auth. This is being tracked in an [issue](https://github.com/charmbracelet/charm/issues/107) though and is definitely something the whole team wants to see implemented
